Assignment 2.Rmd

```diff
@@ -87,13 +98,92 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.widt
 
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
 
+```{r}
+# Count the number of disciplinary actions for each student and teacher
+V2.TCH <- count(D1, tid)
+V2.ST <- count(D1, stid)
+
+#Change the variable names and combine the two sets of numbers
+names(V2.TCH) <- c("id", "number")
+names(V2.ST) <- c("id", "number")
+number <- bind_rows(V2.TCH, V2.ST)
+
+#Add the numbers to VERTEX
+VERTEX <- merge(VERTEX, number)
+
+g2 <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
+
+#Change the vertex size so it reflects the number of disciplinary actions received or given out. Labels are placed above the vertices so they are easier to read. 
+plot(g2, vertex.size=VERTEX$number, edge.width=EDGE$count, vertex.label.cex=0.8, vertex.label.dist=1, vertex.label.degree = -pi/2, edge.arrow.size = 0.8, asp=0)
+```
 
 
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
 
+```{r}
+library(tidyr)
+library(readr)
+hudk4050_classes <- read_csv("hudk4050-classes.csv")
+
+#Select the variables needed, delete the first two rows, and rename the variables
+D3 <- select(hudk4050_classes, Q10:Q7)
+D3 <- D3[-(1:2), ]
+names(D3) <- c("id", "class1", "class2", "class3", "class4", "class5", "class6")
+
+#Gather all the classes into one column, remove the NAs, and add a variable "number" which is the numeric part of the course name. 
+df3 <- gather(D3, "enrolled", "class", 2:7, na.rm = TRUE)
+df3$number <- gsub("[^[:digit:]]", "", df3$class)
+
+#Compare "number" and class name, and rename the duplicates. 
+df3$number[df3$class == "HUDM 4050"] <- "40502"
```

Comment:
Another way is to do the following: `df3$class = gsub(" ", "", df3$class)` to remove the space between the department and the course number.